### ISSUE TITLE
[FR] Allow named container for parameters when converting to `DEProblem`

### ISSUE THREAD

**Issue by Datseris** [SciML/ModelingToolkit.jl#2195]
I have just started using ModelingToolkit.jl with DynamicalSystems.jl. The library seems fantastic. In DynamicalSystems.jl we work directly from the `ODEProblem`, similarly to the `solve` function, hence after the modeled system is built, it is cast into an `ODEProblem`. I'm building a decently complicated model with ~20-100 or so parameters. All of them are specifically named and given a default value using the `@parameters` macro. I've noticed that after convertion to an `ODEProblem` the names dissapear and the parameters are cast into a standard `Vector`:

```julia
julia> parameters(sys)
10-element Vector{SymbolicUtils.BasicSymbolic{Real}}:
 τ_T
 τ_α
 α_tanh_low
 α_tanh_Tref
 α_tanh_smoothness
 α_tanh_high
 ε_tanh_smoothness
 ε_tanh_low
 ε_tanh_Tref
 ε_tanh_high

julia> prob = ODEProblem(sys, nothing, (0.0, 1.0));

julia> prob.p
10-element Vector{Float64}:
   5.0e8
   5.0e9
   0.7
 274.5
   0.1
   0.289
   0.5
   0.5
 288.0
   0.41
```

it would be great if the `ODEProblem(sys::ODESystem, ...)` allows for a way to cast the parameters to a named container. This way, it is much simpler to later down the line do parameter-related analysis with DynamicalSystems.jl or other libraries.

**Comment by Datseris** [comment]
Hm, I was also inspecting how `sol[sym]` works if `sym` is a parameter or a variable that gets eliminated, i.e., an "observed quantity". The code comes from SciMLBase.jl: 
```julia
            if has_sys(A.prob.f) && is_indep_sym(A.prob.f.sys, sym) ||
               Symbol(sym) == getindepsym(A)
                return A.t
            elseif has_sys(A.prob.f) && is_param_sym(A.prob.f.sys, sym)
                return A.prob.p[param_sym_to_index(A.prob.f.sys, sym)]
            elseif has_paramsyms(A.prob.f) && Symbol(sym) in getparamsyms(A)
                return A.prob.p[findfirst(x -> isequal(x, Symbol(sym)), getparamsyms(A))]
            else
                if (sym isa Symbol) && has_sys(A.prob.f)
                    if hasproperty(A.prob.f.sys, sym)
                        return observed(A, getproperty(A.prob.f.sys, sym), :)
                    else
                        error("Tried to index solution with a Symbol that was not found in the system using `getproperty`.")
                    end
                else
                    return observed(A, sym, :)
                end
            end
```

and isn't too hard to parse, so I could alter it so that a similar code could be used with objects that aren't a DiffEqSolution.

**Comment by ChrisRackauckas** [comment]
This should be handled by SII stuff now. Let me know if I misread.
